src/lib/entities/audio/model/audio-instance.ts

```diff
@@ -11,8 +11,19 @@ export const paused = writable(true);
 export const buffered = writable<SvelteTimeRanges>([]);
 
 export const play = (_src?: string) => {
-  if (_src) src.set(_src);
-  paused.set(false);
+  if (_src) {
+    pause();
+    duration.set(0);
```

Comment:
issue: I don't quite understand this `if` block. Could you please write a comment here explaining why this stuff is needed?

Reply:
Setting `duration` to 0 and subscribing to it helps with catching the moment when the the URL takes effect thus "populatnig" the writables with data. And upon that moment, when the data is loaded and the player is ready, I call play.

Reply:
It is of course a hack. But the fact that the audio stops after URL change also seems to be a bug. So I hack the bug.

Reply:
I meant to add a comment to the source code for everyone to read in the future